### DISCUSSION TITLE
Roles Enums

### DISCUSSION THREAD

**Discussion by jtauber** [bible-technology/scripture-burrito#262] [Next]
This is the discussion for #248.

*Roles* are a way of calling out specific ingredients in a burrito of a particular flavour. Because ingredient naming / location is not enforced, *roles* are important in identifying what a particular ingredient is _for_.

For example, a `textTranslation` burrito might need to contain a file for locale data. The `localedata` role can be assigned to that particular ingredient.

We still need to add `versification` role to indicating versification delta file among ingredients.

We need to document how `x-` roles work.

Current roles in schema (there is currently no flavour-restriction enforced, nor uniqueness):

## USX Peripheral Label

- abbreviations
- alphacontents
- chron
- cnc
- contents
- cover
- foreword
- glo
- halftitle
- imprimatur
- lxxquotes
- maps
- measures
- ndx
- preface
- promo
- pubdata
- spine
- tdx
- title

## Story-Based / Video Works

- background
- bridge
- credits
- diagram
- gloss
- illustration
- introduction
- scripture
- teaching
- timing

## Print-on-Demand Content

- body
- thumbnail

## textTranslation

- localedata

## Label for Story-Type Units

- unit XXX

## Label for Dictionary-Type Resources

- name XXX
- sign XXX
- word XXX
- concept XXX
- place XXX

**Comment by jag3773** [comment]
Need to refer to USFM and USX peripheral pages for documentation on what most of these mean: https://ubsicap.github.io/usx/v3.0.0/peripherals.html, https://ubsicap.github.io/usfm/peripherals/index.html.
